library/cipher_wrap.h

```diff
@@ -36,6 +36,42 @@
 extern "C" {
 #endif
 
+/* Support for GCM either through Mbed TLS SW implementation or PSA */
+#if defined(MBEDTLS_GCM_C) || \
+    (defined(MBEDTLS_USE_PSA_CRYPTO) && defined(PSA_WANT_ALG_GCM))
+#define MBEDTLS_CIPHER_HAVE_GCM
```

Comment:
Suggestion for improving the name: `MBEDTLS_CIPHER_HAVE_GCM_USE_PSA`. Not ideal, as it doesn't clearly reflect how this related to `USE_PSA`, but at least if gives a hint that it's somehow related to `USE_PSA`...

Or perhaps go with a longer name: `MBEDTLS_CIPHER_HAVE_GCM_VIA_LEGACY_OR_USE_PSA`? 

Wdyt?

Reply:
I would vote for the longer name as it is much clearer personally

Reply:
I agree, albeit longer it's clearer.